ThreadPool.h

```diff
@@ -62,25 +81,34 @@ inline void ThreadPool::setup(std::size_t threads, F&& initialize) {
         task();
       }
     });
+
+    pthread_attr_t pthread_attribute;
+    pthread_attr_init(&pthread_attribute);
```

Comment:
`pthread_attr_init`, unless set explicitly about the stack size, will have different value depending on the platform